fix GitPython error: len() mismatch in _get_fetch_info_from_stderr

We're seeing this a lot on production. I suspect it may be fixed by a newer version of GitPython. We're using `0.3.2.RC1` and the current release is `0.3.6`. Our issue may be the one described here:

https://github.com/gitpython-developers/GitPython/issues/301

- [ ] reproduce error on dev so we can test
- [x] try upgrading GitPython and see if that fixes it

Expanding the error a bit, I see this doubling-up:

```
AssertionError: len([

    "76bd06d46bdf0a1645e6e47563004780906aface\tnot-for-merge\tbranch 'jvmhrgssl' of /var/opt/chime/data/default/repo\n"
    "1471df09c0551d4a667de02a39bd07e8c573a7d9\tnot-for-merge\tbranch 'master' of /var/opt/chime/data/default/repo\n"
    "89e6df27bb3a6954d9707d0eb9b42df100487330\tnot-for-merge\tbranch 'mtlrpnppq' of /var/opt/chime/data/default/repo\n"
    "1f6201334ff50ac150632a1bc52d9a5c77bb9c04\tnot-for-merge\tbranch 'pxhwsrhtp' of /var/opt/chime/data/default/repo\n"
    "fa0b164f1398819d0ec082f1509582383b89d34f\tnot-for-merge\tbranch 'qsrwqlsjv' of /var/opt/chime/data/default/repo\n"

    "76bd06d46bdf0a1645e6e47563004780906aface\tnot-for-merge\tbranch 'jvmhrgssl' of /var/opt/chime/data/default/repo\n"
    "1471df09c0551d4a667de02a39bd07e8c573a7d9\tnot-for-merge\tbranch 'master' of /var/opt/chime/data/default/repo\n"
    "89e6df27bb3a6954d9707d0eb9b42df100487330\tnot-for-merge\tbranch 'mtlrpnppq' of /var/opt/chime/data/default/repo\n"
    "1f6201334ff50ac150632a1bc52d9a5c77bb9c04\tnot-for-merge\tbranch 'pxhwsrhtp' of /var/opt/chime/data/default/repo\n"
    "fa0b164f1398819d0ec082f1509582383b89d34f\tnot-for-merge\tbranch 'qsrwqlsjv' of /var/opt/chime/data/default/repo\n"

]) != len([

    ' = [up to date]      jvmhrgssl  -> origin/jvmhrgssl'
    ' = [up to date]      master     -> origin/master'
    ' = [up to date]      mtlrpnppq  -> origin/mtlrpnppq'
    ' = [up to date]      pxhwsrhtp  -> origin/pxhwsrhtp'
    ' = [up to date]      qsrwqlsjv  -> origin/qsrwqlsjv'

])
```

--- a/README.md
+++ b/README.md
@@ -7,30 +7,32 @@ Make an issue: https://github.com/chimecms/chime/issues/new
 
 # Install
 
-If you're a Docker user, you can always download the latest build from the Docker Hub as [chimecms/chime](https://hub.docker.com/r/chimecms/chime). Or if you'd like to work with
-the source:
+To work with the Chime source code:
 
 1. Chime CMS is a Python Flask web application. Follow the instructions on
    [Python Virtual Environments](https://github.com/codeforamerica/howto/blob/master/Python-Virtualenv.md)
    to prepare your Python development space. Make sure to use Python 2.7. Your commands may look something like:
- + sudo pip install virtualenv
- + virtualenv -p /usr/bin/python2.7 .venv
- + source .venv/bin/activate
+ + `sudo pip install virtualenv`
+ + `virtualenv -p /usr/bin/python2.7 .venv`
+ + `source .venv/bin/activate`
 
 2. Install the project requirements: `pip install -r requirements.txt`
 
-3. You will need a bare Github repository in the directory `sample-site` with an initial empty commit
-   (this will become configurable in the future):
- + `mkdir sample-site`
- + `cd sample-site`
- + `git init`
- + `git commit --allow-empty -m "First commit"`
+3. You will need a Github repository in the directory `sample-site` cloned from [chimecms/chime-starter](https://github.com/chimecms/chime-starter):
+ + `git clone --bare --single-branch --branch master https://github.com/chimecms/chime-starter.git sample-site`
+ + `git --git-dir sample-site remote rename origin starter` (ignore the error about _config section 'remote.starter.fetch'_)
 
-4. copy env.sample to .env
+4. copy `env.sample` to `.env`.
 
-5. Run app using [Honcho and the `Procfile`](https://github.com/codeforamerica/howto/blob/master/Procfile.md):
+5. You will also need a working copy of [Jekyll](https://github.com/codeforamerica/howto/blob/master/Jekyll.md):
+ + `sudo apt-get -y install rbenv curl`
+ + `./jekyll/install-jekyll.sh`
 
-        $ honcho start
+6. Run Chime in debug mode using [Honcho and the `Procfile`](https://github.com/codeforamerica/howto/blob/master/Procfile.md):
+
+        $ honcho run python ./run.py
+
+If you're a Docker user, we maintain a public build at the Docker Hub as [chimecms/chime](https://hub.docker.com/r/chimecms/chime).
 
 # Who maintains this?
 

--- a/chime/repo_functions.py
+++ b/chime/repo_functions.py
@@ -514,7 +514,7 @@ def save_working_file(clone, path, message, base_sha, default_branch_name):
     if exists(join(clone.working_dir, path)):
         clone.index.add([path])
 
-    clone.index.commit(message.encode('utf-8'))
+    clone.index.commit(message)
     active_branch_name = clone.active_branch.name
 
     #
@@ -750,7 +750,7 @@ def make_commit_message(subject, body):
 def add_empty_commit(clone, subject, body, push=True):
     ''' Adds a new empty commit with the passed details
     '''
-    clone.index.commit(make_commit_message(subject=subject, body=body).encode('utf-8'))
+    clone.index.commit(make_commit_message(subject=subject, body=body))
     active_branch_name = clone.active_branch.name
 
     #

--- a/chime/view_functions.py
+++ b/chime/view_functions.py
@@ -715,7 +715,7 @@ def make_activity_history(repo):
     history = []
     pattern = re.compile(r'\x00Name: (.*?)\tEmail: (.*?)\tDate: (.*?)\tSubject: (.*?)\tBody: (.*?)\x00', re.DOTALL)
     for log_details in pattern.findall(log):
-        name, email, date, subject, body = tuple([item.decode('utf-8') for item in log_details])
+        name, email, date, subject, body = tuple([item for item in log_details])
         commit_category, commit_type, commit_action = get_commit_classification(subject, body)
         log_item = dict(author_name=name, author_email=email, commit_date=date, commit_subject=subject,
                         commit_body=body, commit_category=commit_category, commit_type=commit_type,

--- a/env.sample
+++ b/env.sample
@@ -27,9 +27,9 @@ SUPPORT_PHONE_NUMBER="{Support Phone Number}"
 #   # Optional for setup-script.py Github repository preparation.
 #   GITHUB_CLIENT_ID="{Github OAuth client ID}"
 #   GITHUB_CLIENT_SECRET="{Github OAuth client secret}"
-#   
-#   # Optional URL base for live running website.
-#   LIVE_SITE_URL="http://127.0.0.1:5001/"
-#   
+
+# Optional URL base for live running website.
+LIVE_SITE_URL="http://127.0.0.1:5001/"
+
 #   # Used to push builds to a remote server
 #   PUBLISH_SERVICE_URL="http://example.org/"

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,5 @@
 PyYAML==3.10
-GitPython==0.3.2.RC1
+GitPython==1.0.1
 Flask==0.10.1
 oauth2>=1.5.211
 requests==2.2.1

